modin/backends/base/query_compiler.py

```diff
@@ -1702,218 +3478,1392 @@ def set_index_names(self, names, axis=0):
+    )
+    def str_startswith(self, pat, na=np.NaN):
+        return StrDefault.register(pandas.Series.str.startswith)(self, pat, na)
+
+    @_doc_str_method(method="strip", params="to_strip : str, optional")
+    def str_strip(self, to_strip=None):
+        return StrDefault.register(pandas.Series.str.strip)(self, to_strip)
+
+    @_doc_str_method(method="swapcase", params="")
+    def str_swapcase(self):
+        return StrDefault.register(pandas.Series.str.swapcase)(self)
+
+    @_doc_str_method(method="title", params="")
+    def str_title(self):
+        return StrDefault.register(pandas.Series.str.title)(self)
+
+    @_doc_str_method(method="translate", params="table : dict")
+    def str_translate(self, table):
+        return StrDefault.register(pandas.Series.str.translate)(self, table)
+
+    @_doc_str_method(method="upper", params="")
+    def str_upper(self):
+        return StrDefault.register(pandas.Series.str.upper)(self)
+
+    @_doc_str_method(
+        method="wrap",
+        params="""width : int
+        **kwargs : kwargs""",
+    )
+    def str_wrap(self, width, **kwargs):
+        return StrDefault.register(pandas.Series.str.wrap)(self, width, **kwargs)
+
+    @_doc_str_method(method="zfill", params="width : int")
+    def str_zfill(self, width):
+        return StrDefault.register(pandas.Series.str.zfill)(self, width)
+
+    # End of Str methods
+
+    # Rolling methods
+
+    # FIXME: most of the rolling/window methods takes *args and **kwargs parameters
+    # which only serves the compatibility with numpy, this behaviour is inherited
+    # from the API level, we should get rid of it.
+
+    @_doc_window_method(
+        result="the result of passed functions",
+        action="apply specified functions",
+        method="aggregate",
+        params="""func : str, dict, callable(pandas.Series) -> scalar, or list of such
+        *args : args
+        **kwargs : kwargs""",
+        build_rulles="udf_aggregation",
+    )
+    def rolling_aggregate(self, rolling_args, func, *args, **kwargs):
+        return RollingDefault.register(pandas.core.window.rolling.Rolling.aggregate)(
+            self, rolling_args, func, *args, **kwargs
+        )
+
+    # FIXME: at the query compiler method `rolling_apply` is an alias for `rolling_aggregate`,
+    # one of these should be removed.
+    @_doc_window_method(
+        result="the result of passed function",
+        action="apply specified function",
+        method="apply",
+        params="""func : callable(pandas.Series) -> scalar
+        raw : bool, default: False
+        engine : None, default: None
+            This parameters serves the compatibility purpose. Always have to be None.
+        engine_kwargs : None, default: None
+            This parameters serves the compatibility purpose. Always have to be None.
+        args : tuple, optional
+        kwargs : dict, optional""",
+        build_rulles="udf_aggregation",
+    )
+    def rolling_apply(
```

Comment:
Need to highlight the difference to `rolling_aggregate` in the summary.

Reply:
There is no much difference to highlight. At the query compiler level `apply` is a special case of `aggregate` (`apply` takes one function to execute, when `aggregate` can take multiple functions). `apply` leaked at the backend level by mistake, because the parameters it provides in addition to `aggregate` make no sense at the query compiler. Added deprecation warning at the doc-string for this function.